submissions/Oleksii_Drohachov/Frogger_task/app.js

```diff
@@ -0,0 +1,157 @@
+const entity = {
+  width: 80,
+  height: 60,
+};
+
+const Enemy = function (yAxisPosition, moveSpeed, width, height) {
+  this.xAxisPosition = -80;
+  this.yAxisPosition = yAxisPosition;
+  this.moveSpeed = moveSpeed;
+  this.width = width;
+  this.height = height;
+  this.sprite = "images/enemy-bug.png";
+};
+
+Enemy.prototype.update = function (dt) {
+  this.xAxisPosition += this.moveSpeed * dt;
+
+  if (this.xAxisPosition >= ctx.canvas.width) {
+    this.xAxisPosition = -80;
+  }
+
+  this.checkCollision();
+};
+
+Enemy.prototype.render = function () {
+  ctx.drawImage(
+    Resources.get(this.sprite),
+    this.xAxisPosition,
+    this.yAxisPosition
+  );
+};
+
+Enemy.prototype.checkCollision = function () {
+  if (
+    player.xAxisPosition < this.xAxisPosition + this.width &&
+    player.width + player.xAxisPosition > this.xAxisPosition &&
+    player.yAxisPosition < this.yAxisPosition + this.height &&
+    player.yAxisPosition + player.height > this.yAxisPosition
+  ) {
+    player.score = 0;
+    scoreTitle.textContent = `Current score is: ${player.score}`;
+    player.respawn();
+  }
+};
+
+const enemyStats = [
+  {
+    yAxisPosition: 60,
+    moveSpeed: 180,
+  },
+  {
+    yAxisPosition: 140,
+    moveSpeed: 140,
+  },
+  {
+    yAxisPosition: 220,
+    moveSpeed: 100,
+  },
+];
+
+const Player = function (xAxisPosition, yAxisPosition) {
+  this.xAxisPosition = xAxisPosition;
+  this.yAxisPosition = yAxisPosition;
+  this.moveXAxis = 100;
+  this.moveYAxis = 90;
+  this.sprite = "images/char-boy.png";
+  this.score = 0;
+  this.scoreMax = 0;
+};
+
+Player.prototype = entity;
+
+Player.prototype.update = function () {
+  if (player.yAxisPosition < 0) {
+    player.respawn();
+    player.win();
```

Comment:
What is the way to address object instance properties and methods?
This is not a rhetoric question. You may also want to review the module's theoretical materials.